ext/date/php_date.h

```diff
@@ -93,6 +93,7 @@ struct _php_period_obj {
 	int               recurrences;
 	bool              initialized;
 	bool              include_start_date;
+	int               include_end_date;
```

Comment:
Should't this be a `bool`? A test for `recurrences` is missing, I think this would result in one too many iterations.

Reply:
@cmb69 I was referring to using `recurrences` and `include_end_date` together which I don't think makes sense as recurrences determine the end date. Unless I'm missing something, I'm not familiar with `DatePeriod` at all.

Reply:
I think the patch was originally made when the `bool include_start_date` was still an `int`.

`recurrences` and `include_end_date` together make no sense, but `recurrences` and `include_start_date` could perhaps. I'll have a look at seeing if anything needs addressing/excluding.

Reply:
I think the patch was originally made when the `bool include_start_date` was still an `int`.

`recurrences` and `include_end_date` together make no sense, but `recurrences` and `include_start_date` could perhaps. I'll have a look at seeing if anything needs addressing/excluding.

Reply:
Now that I have looked this, I am reluctant changing this to a bit mask, due to the negative of "exclude start date". It makes the rest of the code harder to read and understand, so I won't be changing this.

Reply:
I wasn't referring to a bit*mask*, but rather a bit*field*, i.e.
````c
int initialized:1;
int include_start_date:1;
int include_end_date:1;
````

Reply:
Oh :-) — I would have hoped the compiler would do with a bool type. But it doesn't. Changing them to `bool ...:1` does work, and is IMO prefered over `int ...:1`:

```
struct _php_period_obj {
        timelib_time *             start;                /*     0     8 */
        zend_class_entry *         start_ce;             /*     8     8 */
        timelib_time *             current;              /*    16     8 */
        timelib_time *             end;                  /*    24     8 */
        timelib_rel_time *         interval;             /*    32     8 */
        int                        recurrences;          /*    40     4 */
        _Bool                      initialized:1;        /*    44: 0  1 */
        _Bool                      include_start_date:1; /*    44: 1  1 */
        _Bool                      include_end_date:1;   /*    44: 2  1 */

        /* XXX 5 bits hole, try to pack */
        /* XXX 3 bytes hole, try to pack */

        zend_object                std;                  /*    48    56 */

        /* size: 104, cachelines: 2, members: 10 */
        /* sum members: 100, holes: 1, sum holes: 3 */
        /* sum bitfield members: 3 bits, bit holes: 1, sum bit holes: 5 bits */
        /* last cacheline: 40 bytes */
};
```

Reply:
Although it doesn't actually end up saving memory as the zend_object needs to be aligned to an 8 byte boundary ;-)

Reply:
Ah, you're right! (Integrating `recurrences` in the bitfield would be an option, but doesn't look that it's worth it.)